From: Mark Adler <madler@alumni.caltech.edu>
Date: Sun, 2 Feb 2025 21:34:20 +0100
Subject: [PATCH] Fix bug in UZinflate() that incorrectly updated G.incnt.

    Fix bug in UZinflate() that incorrectly updated G.incnt.

    The update assumed a full buffer, which is not always full. This
    could result in a false overlapped element detection when a small
    deflate-compressed file was unzipped using an old zlib. This
    commit remedies that.


Imported from unzip_6.0-28.debian.tar.xz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 inflate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inflate.c b/inflate.c
index f2f6864e1835..2c379991ad09 100644
--- a/inflate.c
+++ b/inflate.c
@@ -700,7 +700,7 @@ int UZinflate(__G__ is_defl64)
       G.dstrm.total_out));
 
     G.inptr = (uch *)G.dstrm.next_in;
-    G.incnt = (G.inbuf + INBUFSIZ) - G.inptr;  /* reset for other routines */
+    G.incnt -= G.inptr - G.inbuf;       /* reset for other routines */
 
 uzinflate_cleanup_exit:
     err = inflateReset(&G.dstrm);
